File: app/java/src/main/java/com/example/production/domain/model/process/LaborHours.java
package com.example.production.domain.model.process;

import com.example.production.domain.model.item.Item;
import com.example.production.domain.model.organization.Department;
import com.example.production.domain.model.organization.Employee;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 工数実績データ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LaborHours {
    private Integer id;
    private String laborHoursNumber;
    private String workOrderNumber;
    private String itemCode;
    private Integer sequence;
    private String processCode;
    private String departmentCode;
    private String employeeCode;
    private LocalDate workDate;
    private BigDecimal hours;
    private String remarks;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // リレーション
    private WorkOrder workOrder;
    private Item item;
    private Process process;
    private Department department;
    private Employee employee;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Process/LaborHours.cs
using ProductionManagement.Domain.Models.Master;

namespace ProductionManagement.Domain.Models.Process;

/// <summary>
/// 工数実績
/// </summary>
public class LaborHours
{
    public int Id { get; set; }
    public required string LaborHoursNumber { get; set; }
    public required string WorkOrderNumber { get; set; }
    public required string ItemCode { get; set; }
    public required int Sequence { get; set; }
    public required string ProcessCode { get; set; }
    public required string DepartmentCode { get; set; }
    public required string EmployeeCode { get; set; }
    public required DateOnly WorkDate { get; set; }
    public required decimal Hours { get; set; }
    public string? Remarks { get; set; }
    public DateTime CreatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }

    // リレーション
    public WorkOrder? WorkOrder { get; set; }
    public Item.Item? Item { get; set; }
    public Process? Process { get; set; }
    public Department? Department { get; set; }
    public Employee? Employee { get; set; }
}
